### PR TITLE
Add "catalog list" command

### LINKED ISSUE #135: Safeguard "clean" Command
## Context
The commodore clean command removes the entire inventory and all components. Changes which are not checked into git will be lost. On a regular catalog compile (if not `--local` mode), the clean command is run first.

The safeguard should interactively ask if changes should really be dropped in case of a dirty workspace. A `--force` flag could be implemented to drop changes without asking first (for example in a CI/CD or test context).

### PR DESCRIPTION
**Pull request by chrisglass**
This commit adds the "commodore catalog list" command that will query
lieutenant for a list of available cluster IDs (and descriptions), and
display the information on stdout.

This partially helps with #135 

## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Update the documentation.
- [ ] Update tests.
- [x] Update the ./CHANGELOG.md.
- [x] Link this PR to related issues.

## Testing done

Unit tests have not been added since the logic is trivial (basically printing out contents of an already-existing API) and consists mostly of stiching side-effects. Mocking side-effects (the click library and lieutenant function call) would result it unit testing a for loop, which I suspect is not very interesting.

### CODE REVIEW

**Comment by srueg** [comment]
Nice addition, thanks for the contribution! 👍 

May I suggest to add a `-o/--output` flag to optionally change the output format. This could include the option `json` to enable structured output to further processing, for example:
```
for catalog in $(commodore catalog list | jq -r '.[]'); do
  commodore catalog compile ${catalog};
done
```

**Review by srueg — APPROVED** [review]

File: docs/modules/ROOT/pages/reference/cli.adoc [hunk 1]
```diff
@@ -92,6 +92,10 @@ See also xref:local-mode.adoc[local mode documentation].
 *--help*::
   Show catalog clean usage and options then exit.
 
+== Catalog Clean
```
Comment (simu) [inline docs/modules/ROOT/pages/reference/cli.adoc:95]: should be `catalog list`

**Review by simu — APPROVED** [review]
LGTM, thanks for the contribution

File: docs/modules/ROOT/pages/reference/cli.adoc (hunk 1 above)
Comment (chrisglass) [reply docs/modules/ROOT/pages/reference/cli.adoc]: Ooops! fixed.

**Pull request merged**